ControlSettings/ControlSettings.cpp

```diff
@@ -248,16 +248,29 @@ namespace Plugin {
                , m_isDisabledHdmiIn4KZoom (false)
     {
         LOGINFO("Entry\n");
-        devicePtr = new WPEFramework::Plugin::TV;
+#ifndef ENABLE_TV_SUPPORT
+        devicePtr = new STB;
+#else
+        devicePtr = new TV;
+#endif
+
 	ControlSettings::_instance = this;
 	InitializeIARM();
 
+        if(devicePtr->isTvSupportEnabled())//TV Specific API
```

Comment:
I observe lots of  code in control setting which has reference to TV ...should be move to TV specific implemetnation

Reply:
As per my previous comment All registartion and Init sequence has been kept in "ControlSettings" Main class.
So only AS can invoke the corresponding API.